core/src/main/java/com/aem/demo/core/models/NewsCardModel.java

```diff
@@ -39,7 +44,62 @@ public class NewsCardModel {
     @Default(values = DEFAULT_IMAGE)
     private String image;
 
+//    private int likes;
+
+    @Inject
+    private SlingHttpServletRequest servletRequest;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+
     public String getPath() {
-        return resource.getPath().replace("/jcr:content","");
+        return resource.getPath().replace("/jcr:content", "");
+    }
+
+    public String getFullPath() {
+        return resource.getPath();
+    }
+
+
+    public int  getLikes() {
+        int likes = 0;
+
+        String rootPath = resource.getPath();
+
+        resource = resourceResolver.getResource(rootPath);
+        ModifiableValueMap properties = resource != null ? resource.adaptTo(ModifiableValueMap.class) : null;
+
+        if (properties != null) {
+            likes = resource.adaptTo(ValueMap.class).get("likes", 0);
+            properties.put("likes", likes + 1);
+        }
+        try {
+            resourceResolver.commit();
+        } catch (PersistenceException e) {
+            log.info(e.getMessage());
+        }
+        return likes;
+    }
+
+    public int  getDisLikes() {
```

Comment:
This method is almost the exact copypaste of the previous one. The only difference is just the name of the property: "likes/dislikes".
Consider creating a common method for both the likes and dislikes, with the name of the property passed as a method argument. Surely, after all the refactoring that was mentioned in the previous comments.